### PR TITLE
Remove prompting from optimize workflow

### PR DESCRIPTION
**Pull request by sergeichestakov**
After some user feedback, I think it might be best to remove the prompts from the `expo optimize` workflow (including via `expo publish`) in favor of warnings.

The prompt in `expo optimize` was to let the user know that their assets were going to be overwritten and ask if they want to save.a backup. However, given that the images will most likely look the same this shouldn't be a problem. In the off chance that they don't and the quality is noticeably worse, they can always restore them using git since Expo projects are almost always git projects (unless the user deleted the .git folder which is unlikely). I think we should also discourage saving backups as that clutters the file system and is most likely unnecessary.

The prompt during `expo publish` was to let the user know about how to optimize and give them the chance to do so right then and there. However, a warning will accomplish the same thing as far as discoverability and be less annoying since it prevents publishing until the user responds (also works better in non-interactive settings like CI). If they do want to optimize, they can run the command after, double check the images look good, and then republish.

I also made some minor changes including checking to make sure the amount saved is not undefined before logging it (I was getting `saved NaN undefined` if I deleted the `.expo-shared` folder and tried to re-optimize).

### CODE REVIEW

File: packages/xdl/src/Project.js [hunk 1]
```diff
@@ -2094,13 +2094,14 @@ export async function optimizeAsync(projectRoot: string = './', options: Object
   });
 
   // Check for custom quality value
+  const defaultQuality = 60;
   const { quality: strQuality, include, exclude, save } = options;
   const quality = Number(strQuality);
   const validQuality = Number.isInteger(quality) && quality > 0 && quality <= 100;
   if (strQuality !== undefined && !validQuality) {
-    logger.global.warn('Invalid quality entered. Using default of 60.');
+    logger.global.warn(`Invalid quality entered. Using default of ${defaultQuality}.`);
```
[the remark this answers is not in the document]
Comment (sergeichestakov) [reply packages/xdl/src/Project.js]: Good call. I moved the validation to `expo-cli` and now throw an error instead in bfce457. Let me know if that looks ok.

**Comment by esamelson** [comment]
code looks good, just not totally sure I agree with this:

> After some user feedback, I think it might be best to remove the prompts from the expo optimize workflow (including via expo publish) in favor of warnings.

Just want to point out that one person complaining is not necessarily an indicator that we should hop to change this right away. (i.e. plenty of other people have clearly come across this and not complained!) Warnings may be less annoying, but also because of that people are much more likely to just ignore them forever. If a big reason for having this is to have more optimized assets on our server then I would argue that having the interactive prompt is the best way to do that. Thoughts? @sergeichestakov @ide @fson

**Comment by sergeichestakov** [comment]
Sorry, I should clarify that I'm not just proposing this because a handful of people requested it, but my own personal opinion on the workflow has changed after using it for longer and in different scenarios.

So I think it might be more convenient for most people to have a warning and have it still be obvious enough that it will at least spike people's interest and incentivize them to run the command themselves (at the very least to make the warning go away).

**Comment by ide** [comment]
I think we could do a mix of things (not to compromise but if we think it is the best approach) e.g. show the prompt the first time and then show the warning afterwards.

File: packages/xdl/src/Project.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (fson) [reply packages/xdl/src/Project.js]: That looks good to me!

**Comment by fson** [comment]
I think a prompt as a user interface element is similar to an alert, in that it disrupts the user flow and blocks the user interface until the user takes an action. Alerts should only reserved for cases when continuing without user confirmation is not possible. For example, the 
[Apple Human Interface Guidelines has the following note on alerts](https://developer.apple.com/design/human-interface-guidelines/macos/windows-and-views/alerts/)): 
> **Minimize alerts.** Alerts disrupt the user experience and should only be used in important situations like confirming purchases and destructive actions (such as deletions), or notifying people about problems. The infrequency of alerts helps ensure that people take them seriously."

In my opinion, not having optimized the assets is not an error condition that should block the normal development flow, but instead something we should warn about and let the user fix it when convenient for them.

Ideally we would constantly show a non-blocking warning element that would go away when the assets have been optimized. It turns out in Expo Dev Tools we already have a UI element like this: the Issues tab.
Screenshot 2019-05-29 at 18 58 18
Any error/warning that is logged with an additional `id` becomes a sticky notification that can be cleared with `ProjectUtils.clearNotification`. ([Example of this](https://github.com/expo/expo-cli/blob/59a4abb14e1edb80bd2fd23be4d7a38285df6810/packages/xdl/src/project/Doctor.js#L248-L257)). I think adding the warning about optimizing assets to the Issues tab would be a good addition and also somewhat address @esamelson's concern about people ignoring the warning.

Later, I think adding a "Fix" button in the warning would be a nice extra. Pushing the button would immediately optimize the assets. (This kind of UI is something we envisioned when building this feature in Dev Tools, although right now we have no issues that have a "Fix" button.) However, this would require some additional UI work, so I think right now the best we can do is make the UI non-blocking, like this PR does, and additionally make the warning a sticky notification, and at some point later, possibly enhance the experience by adding a "Fix" button.

**Comment by sergeichestakov** [comment]
That's a good point Ville. I added a warning for this issue to dev tools in 2ac6f0d. If that looks good to you guys I can merge this.

**Comment by fson** [comment]
Looks great. so I'm going to merge this now.

Thanks @sergeichestakov!

**Comment by tim92109** [comment]
Thanks.  Now have to start over.  "Optimize assets"?  Maybe it should say "break my app"

**Pull request merged**